gluten-core/src/main/scala/io/glutenproject/expression/DateTimeExpressionsTransformer.scala

```diff
@@ -108,35 +111,40 @@ class DayOfMonthTransformer(child: Expression) extends DayOfMonth(child) with
   override def getDataType: DataType = dataType
 
   override def getChild: Expression = child
-
-  override def doTransform(args: java.lang.Object): ExpressionNode = {
-    val childNode = child.asInstanceOf[ExpressionTransformer].doTransform(args)
-    if (!childNode.isInstanceOf[ExpressionNode]) {
-      throw new UnsupportedOperationException(s"Not supported yet.")
-    }
-
-    val functionMap = args.asInstanceOf[java.util.HashMap[String, java.lang.Long]]
-    val functionId = ExpressionBuilder.newScalarFunction(functionMap,
-      ConverterUtils.makeFuncName(ConverterUtils.DAY_OF_MONTH,
-        Seq(child.dataType), FunctionConfig.OPT))
-    val expressionNodes = Lists.newArrayList(childNode)
-    val typeNode = ConverterUtils.getTypeNode(getDataType, true)
-    ExpressionBuilder.makeScalarFunction(functionId, expressionNodes, typeNode)
-  }
 }
 
 class DayOfYearTransformer(child: Expression) extends DayOfYear(child) with
   ExtractDateTransformer {
-  override def getField: String = "DAYOFYEAR"
+  override def getField: String = "DAY_OF_YEAR"
 
   override def getDataType: DataType = dataType
 
   override def getChild: Expression = child
+
 }
 
 class DayOfWeekTransformer(child: Expression) extends DayOfWeek(child) with
   ExtractDateTransformer {
-  override def getField: String = "DAYOFWEEK"
+  override def getField: String = "DAY_OF_WEEK"
+
+  override def getDataType: DataType = dataType
+
+  override def getChild: Expression = child
+
+}
+
+class WeekDayTransformer(child: Expression) extends WeekDay(child) with
+  ExtractDateTransformer {
+  override def getField: String = "DAY_OF_WEEK"
```

Comment:
The return values of these two date function are different, is it ok to use the same field name?

Reply:
Updated.